Tests/AuthFoundationTests/OAuth2ClientTests.swift

```diff
@@ -143,7 +143,91 @@ final class OAuth2ClientTests: XCTestCase {
         XCTAssertEqual(userInfo?.subject, "00uid4BxXw6I6TV4m0g3")
     }
 
-    func testIntrospect() throws {
+    func testIntrospectTokenRequest() throws {
+        let openIdConfiguration = try OpenIdConfiguration.jsonDecoder.decode(
+            OpenIdConfiguration.self,
+            from: try data(from: .module,
+                           for: "openid-configuration",
+                           in: "MockResponses"))
+        let request = try Token.IntrospectRequest(openIdConfiguration: openIdConfiguration,
+                                                  token: token,
+                                                  type: .accessToken)
+        XCTAssertNil(request.authorization)
+        XCTAssertEqual((request.bodyParameters?["client_id"] as? String), token.context.configuration.clientId)
```

Comment:
I'd suggest asserting something concrete, rather than something like `token.context.configuration.clientId`.

It's harder to know what's expected when it's coming from somewhere else, and it's also easy to break your expectations without knowing it.

Reply:
Good callout!. I've updated this.